### PR TITLE
#8519: add text option only when select is not an option in non-expression contexts

### LINKED ISSUE #8519: Broken field type toggle during mod activation
## Describe the bug

- Selecting `ABC` in the field type toggle during mod activation for `@pixies/template/translation` doesn't do anything
- Should `ABC` even be an option for the field given the schema type? Need to confirm the expected behavior
- https://pixiebrix.slack.com/archives/C0436P48QHY/p1716861180646839

## To Reproduce

1. Activate `@pixies/template/translation` in the Extension Console
1. Reactivate `@pixies/template/translation` in the sidebar

## Options Schema

The field uses the oneOf const/title options formulation:

```yaml
  schema:
    type: object
    $schema: https://json-schema.org/draft/2019-09/schema#
    required:
      - defaultLanguage
    properties:
      defaultLanguage:
        type: string
        oneOf:
          - const: am
            title: 🇪🇹 Amharic
```

## Actual behavior

- Field shows a type toggle
- Selecting `ABC` does nothing and there's no error in the console

## Expected behavior

- If `ABC` is a valid option, field should switch to text entry when selected
- If `ABC` is not a valid option, it's not included in the field toggle dropdown. If there's a single valid field type, show the field toggle as disabled.
- TODO: determine expected available field type toggles for `@pixies/template/translation`
    - Should only ever be a select per @BrandonPxBx 

## Screenshots/Loom

image

### PR DESCRIPTION
**Pull request by grahamlangford**
## What does this PR do?

- Fixes #8519 

## Discussion

- NOTE: schema with options (examples, enum, oneOf, etc) will no longer display text inputs as an option

## Demo

https://www.loom.com/share/66cdd7a0f6154140acbaf4eefeb37a76

### CSS fix
image

## Checklist

- [x] Add jest or playwright tests and/or storybook stories
- [x] Designate a primary reviewer @fungairino

### CODE REVIEW

**Review by BrandonPxBx — APPROVED** [review]

**Comment by twschiller** [comment]
@grahamlangford @BrandonPxBx did we have a regression on the right padding on the form fields in sidebar activation?

image

File: src/components/fields/schemaFields/getToggleOptions.tsx [hunk 1]
```diff
@@ -307,10 +307,14 @@ export function getToggleOptions({
...
           : null,
     });
+    handleVarOption();
   }
 
   // Select fields will match the basic string check also
-  if (fieldSchema.type === "string" || anyType) {
+  if (
+    (fieldSchema.type === "string" || anyType) &&
+    !isSelectField(fieldSchema)
```
Comment (twschiller) [inline src/components/fields/schemaFields/getToggleOptions.tsx:316]: @grahamlangford to confirm, this isn't getting rid of var option on the page editor?

**Comment by twschiller** [comment]
@grahamlangford what's the diagnosis on the root cause of the issue in the ticket? From the code did it look like we never intended to support both select and text? 

My concern is text input can be helpful for conditionally setting the value in the Page Editor (when configuring a brick). But agree for mod options it's not valuable because the value cannot be a text template

File: src/components/fields/schemaFields/getToggleOptions.tsx (hunk 1 above)
Comment (grahamlangford) [reply src/components/fields/schemaFields/getToggleOptions.tsx:316]: The `var` option is handled in the change a couple of lines above

**Comment by grahamlangford** [comment]
> @grahamlangford what's the diagnosis on the root cause of the issue in the ticket? From the code did it look like we never intended to support both select and text?
> 
> My concern is text input can be useful for conditionally setting the value in the Page Editor (when configuring a brick). But agree for mod options (and integration configuration) it's not valuable because the value cannot be a text template
> 
> During activation, where is the logic that determines that we don't show variable as a toggle option? We could use that same location to determine text entry is not a valid entry mod for selectable fields during activation / integration configurations
> 
> An example would be the highlight text brick, which has `examples` set for the color. In the Page Editor, the text template entry works:
> 

Honestly, the current solution I'm proposing in this Draft PR is probably the wrong way to go. It's resulting in cascading problems in multiple areas (including Google Sheets brick). Might have to go back to the drawing board.

File: src/components/fields/schemaFields/BasicSchemaField.test.tsx [hunk 2]
```diff
@@ -70,7 +70,8 @@ describe("option mode switching", () => {
     expectToggleMode(container, "Text");
   });
 
-  test("switches automatically from variable to text when @ removed - enum field", async () => {
+  // TODO: Determine whether this is correct behavior
```
Comment (fungairino) [inline src/components/fields/schemaFields/BasicSchemaField.test.tsx:73]: Is there a corresponding UX ticket?

File: src/sidebar/activateMod/ActivateModPanel.module.scss [hunk 3]
```diff
@@ -71,8 +71,8 @@
 }
 
 .optionsBody {
-  margin-left: -20px;
-  margin-right: -20px;
+  padding-left: 0px;
+  padding-right: 0px;
```
Comment (fungairino) [inline src/sidebar/activateMod/ActivateModPanel.module.scss:75]: nit: what's the default? Can we just remove these attributes?

File: src/components/fields/schemaFields/BasicSchemaField.test.tsx (hunk 2 above)
Comment (grahamlangford) [reply src/components/fields/schemaFields/BasicSchemaField.test.tsx:73]: Should have been removed. Confirmed with @BrandonPxBx

File: src/sidebar/activateMod/ActivateModPanel.module.scss (hunk 3 above)
Comment (grahamlangford) [reply src/sidebar/activateMod/ActivateModPanel.module.scss:75]: Now we can't. This is overriding Bootstrap. I'll leave a comment

**Pull request merged**